graylog2-server/src/main/java/org/graylog2/security/MongoDbSessionDAO.java

```diff
@@ -60,10 +60,7 @@ protected Serializable doCreate(Session session) {
         fields.put("last_access_time", session.getLastAccessTime());
         fields.put("timeout", session.getTimeout());
         Map<String, Object> attributes = Maps.newHashMap();
-        for (Object key : session.getAttributeKeys()) {
-            attributes.put(key.toString(), session.getAttribute(key));
-        }
-        fields.put("attributes", attributes);
+        fields.put("attributes", MongoDbSession.transformAttributes(attributes));
```

Comment:
Aren't we missing the session attributes now? We are transforming an empty map now. Or are there no session attributes on session creation?

Reply:
This is a mistake, thanks for catching that. But also, when I looked into it, there actually were not session attributes at creation time present.